agent/lib_aws_sdk_php.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Functions relating to instrumenting the AWS-SDK-PHP.
+ * https://github.com/aws/aws-sdk-php
+ */
+#include "php_agent.h"
+#include "php_wrapper.h"
+#include "fw_hooks.h"
+#include "fw_support.h"
+#include "util_logging.h"
+#include "php_call.h"
+#include "lib_aws_sdk_php.h"
+
+#define PHP_PACKAGE_NAME "aws/aws-sdk-php"
+#define PHP_AWS_CLASS_PREFIX "Supportability/library/aws/aws-sdk-php/"
+
+/*
+ * In a normal course of events, the following line will always work
+ * zend_eval_string("Aws\\Sdk::VERSION;", &retval, "Get AWS Version")
+ * By the time we have detected the existence of the aws-sdk-php and with
+ * default composer profject settings, it callable even from
+ * nr_aws_sdk_php_enable which will automatically load the class if it isn't
+ * loaded yet and then evaluate the string. However, in the rare case that files
+ * are not loaded via autoloader and/or have non-default composer classload
+ * settings, if the class is not found, PHP 8.2+ will generate a fatal
+ * unrecoverable uncatchable error error whenever it cannot find a class. While
+ * calling this from nr_aws_sdk_php_enable would have been great and would allow
+ * the sdk version value to be set only once, to avoid the very unlikely but not
+ * impossible fatal error, this will be called from the
+ * "Aws\\ClientResolver::_apply_user_agent" wrapper which GUARANTEES that
+ * aws/sdk exists and is already loaded.
+ *
+ *
+ * Additionally given that aws-sdk-php is currently detected from the
+ * AwsClient.php file, this method will always be called when a client is
+ * created unlike Sdk::construct which doesn't show with PHP 8.2+.
+ *
+ * Using Aws/Sdk::__construct for version is currently nonviable as it is
+ * unreliable as a version determiner.
+ * Having separate functionality to extract from Aws/Sdk::__construct
+ * is both not required and is redundant and causes additional overhead and
+ * so only one function is needed to extract version.
+ *
+ * Aws\\ClientResolver::_apply_user_agent a reliable function as it is
+ * always called on client initialization since it is key to populating
+ * the request headers, and it loads Sdk by default.
+ *
+ * Concerns about future/past proofing to the checking prioritized the following
+ * implementation vs using the eval method.
+ */
+extern void lib_aws_sdk_php_handle_version() {
```

Comment:
`extern` keyword is not needed when defining the function:
```suggestion
void lib_aws_sdk_php_handle_version() {
```

Reply:
Please see: [7d98b70](https://github.com/newrelic/newrelic-php-agent/pull/932/commits/7d98b70ca2e9ada5dc79739fefa63e535ae6220d)